### PR TITLE
#167867741 User/Admin should be able to view one trip (Admin should be able to cancel/delete)

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
User/Admin should be able to view one trip 
Admin should be able to cancel a trip
Admin should be able to update a trip
Admin should be able to delete a trip

#### Description of Task to be completed?
Create SQL query in tripModel to get one trip from the db
Create SQL query in tripModel to  cancel a trip from the db
Create SQL query in tripModel to update one trip from the db

Create tests for get-onetrip, cancel-trip, update-trip
Implemented the filter trips by origin/destination

#### How should this be manually tested?
Clone the repo to your machine and run:
* `npm install`.
* `npm run dev`.
* create a postgres database named `dbname`.
* Download Postman.
* In Postman, create a new GET request tab and enter the following URL: 
* For one trip `http://localhost:3000/api/v2/trips/1`.
* Click `Send`

* You should have s similar response to the following:  
{
    "status": 200,
    "message": "Trips Successfully Fetched",
    "data": [
        {
            "id": 1,
            "seating_capacity": 45,
            "bus_license_number": "KCK 469",
            "origin": "Kigali",
            "destination": "Nairobi",
            "trip_date": "10-12-2019",
            "fare": 5000,
            "status": 1
        }
    ]
}
* For cancel a trip, admin goes to PATCH `http://localhost:3000/api/v2/trips/1/cancel`
*Response {
    "status": 200,
    "message": "Trip Successfully Cancelled",
    "data": {
        "id": 1,
        "seating_capacity": 45,
        "bus_license_number": "KCK 469",
        "origin": "Kigali",
        "destination": "Nairobi",
        "trip_date": "10-12-2019",
        "fare": 10000,
        "status": 9
    }
}
* For deleting a trip, admin goes to DELETE `http://localhost:3000/api/v2/trips/1`
Response 204 No content

* For update a trip, admin goes to PATCH `http://localhost:3000/api/v2/trips/1`

* Send  
{
"fare": 15000
}
*Response {
    "status": 200,
    "message": "Trip Successfully Upadated",
    "data": {
        "id": 1,
        "seating_capacity": 45,
        "bus_license_number": "KCK 469",
        "origin": "Kigali",
        "destination": "Nairobi",
        "trip_date": "10-12-2019",
        "fare": 10000,
        "status": 9
    }
}

#### What are the relevant pivotal tracker stories?
[#167876336](https://www.pivotaltracker.com/story/show/167876336)
[#167876387](https://www.pivotaltracker.com/story/show/167876387)
[#167876334](https://www.pivotaltracker.com/story/show/167876334)
[#167876516](https://www.pivotaltracker.com/story/show/167876516)

#### Screenshots (if appropriate)
admin-cancel-trip
admin-delete-trip
get-trip-admin

### CODE REVIEW

**Comment by ekumamait** [comment]
Nice work @nignanthomas however, it looks like you are not returning any response after the admin has deleted a trip but It would be correct to return the `status code` and `a message` in the response. Also, am wondering why we still need a token to get a specific trip and it looks like you are using admin to fetch that trip, I would like to know why?

File: server/v2/helpers/validators.js [hunk 1]
```diff
@@ -22,9 +22,9 @@ const validateNewTrip = Joi.object().keys({
 });
 
 const validateUpdateTrip = Joi.object().keys({
-  seating_capacity: Joi.number(),
-  bus_license_number: Joi.string().trim(),
-  trip_date: Joi.date().min('now').raw(),
+  // seating_capacity: Joi.number(),
```
Comment (ekumamait) [inline server/v2/helpers/validators.js]: @nignanthomas if we are not using this code maybe we should remove it

**Comment by nignanthomas** [comment]
> Nice work @nignanthomas however, it looks like you are not returning any response after the admin has deleted a trip but It would be correct to return the `status code` and `a message` in the response. Also, am wondering why we still need a token to get a specific trip and it looks like you are using admin to fetch that trip, I would like to know why?

Thanks @ekumamait 
I tried returning a custom message but it wasn't and I've been told that that's the default behavior of 204, not to display anything.
When accessing the route with admin token, the admin can view all (active + canceled), while regular users only see active.

**Review by ekumamait — COMMENTED** [review]
Am still wondering why we have commented code in the codebase.

**Pull request merged**